server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -251,31 +267,50 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
                         hypervisorId.equalsIgnoreCase(knownHost.getHypervisorId().getHypervisorId()) &&
                         knownHost.getHypervisorId().getReporterId() != null &&
                         !jobReporterId.equalsIgnoreCase(knownHost.getHypervisorId().getReporterId())) {
+
                         log.debug("Reporter changed for Hypervisor {} of Owner {} from {} to {}",
                             hypervisorId, ownerKey, knownHost.getHypervisorId().getReporterId(),
                             jobReporterId);
                     }
-                    if (consumerResource.performConsumerUpdates(incoming, knownHost, guestConsumersMap,
-                        false)) {
-                        consumerCurator.update(knownHost);
-                        result.updated(updateCheckinTime(knownHost));
+
+                    boolean changesMade = consumerResource.checkForFactsUpdate(knownHost, incoming);
+                    changesMade = consumerResource.checkForGuestsUpdate(knownHost, incoming) || changesMade;
+
+                     if (changesMade) {
+                         complianceRules.getStatus(knownHost, null, false, false);
+                         result.updated(knownHost);
                     }
                     else {
-                        result.unchanged(updateCheckinTime(knownHost));
+                         result.unchanged(knownHost);
                     }
                 }
+
                 // update reporter id if it changed
                 if (jobReporterId != null && reportedOnConsumer != null &&
                     reportedOnConsumer.getHypervisorId() != null &&
                     (reportedOnConsumer.getHypervisorId().getReporterId() == null ||
                     !jobReporterId.contentEquals(reportedOnConsumer.getHypervisorId().getReporterId()))) {
+
                     reportedOnConsumer.getHypervisorId().setReporterId(jobReporterId);
                 }
                 else if (jobReporterId == null) {
                     log.debug("hypervisor checkin reported asynchronously without reporter id " +
                         "for hypervisor:{} of owner:{}", hypervisorId, ownerKey);
                 }
             }
+
+            Date now = new Date();
+            java.util.Collection<Consumer> c = hypervisorConsumersMap.getConsumers();
```

Comment:
Another place where we shouldn't use the fully-qualified class here -- should do a proper import.